src/shared/ui/Banners/RebrandBanner/index.tsx

```diff
@@ -0,0 +1,119 @@
+import { useEffect, useState } from 'react';
+
+import { Box, Collapse } from '@mui/material';
+import { Trans } from 'react-i18next';
+import { useLocation } from 'react-router-dom';
+
+import { Banner, BannerProps } from '../Banner';
+import { StyledImg } from './RebrandBanner.styles';
+
+import curiousIcon from '~/assets/curious_icon--white.png';
+import { userModel } from '~/entities/user';
+
+/**
+ * Returns a unique key for the rebrand banner dismiss state
+ * Otherwise, it creates a user-only key for use on the auth screen
+ */
+export const getDismissedKey = (userId: string) => `rebrand-banner-dismissed-${userId}`;
+
+export const GLOBAL_DISMISSED_KEY = 'rebrand-banner-dismissed-global';
+
+const NON_AUTH_DISPLAY_ROUTES = [
+  /^\/login$/,
+  /^\/signup$/,
+  /^\/forgotpassword$/,
+  /^\/password-recovery$/,
+];
+const AUTH_DISPLAY_ROUTES = [
+  /^\/protected\/(profile|settings)$/,
+  /^\/protected\/applets(\/[^/]+)?$/,
+];
+
+export const RebrandBanner = (props: BannerProps) => {
+  const location = useLocation();
+  const { isAuthorized, user } = userModel.hooks.useAuthorization();
+  const userId = user?.id;
+
+  const [isRebrandBannerActive, setIsRebrandBannerActive] = useState(true);
+  const [isCollapsing, setIsCollapsing] = useState(false);
+
+  const handleOnExited = () => {
+    if (isCollapsing) {
+      setIsCollapsing(false);
+      setIsRebrandBannerActive(false);
+    }
+  };
+
+  useEffect(() => {
+    // For auth screen or when user is not logged in yet
+    if (!isAuthorized) {
+      const globalDismissed = localStorage.getItem(GLOBAL_DISMISSED_KEY);
+      setIsRebrandBannerActive(!globalDismissed);
+      return;
+    }
+
+    // For logged-in users
+    if (userId) {
+      const userDismissed = localStorage.getItem(getDismissedKey(userId));
+      setIsRebrandBannerActive(!userDismissed);
+    }
+  }, [isAuthorized, userId]);
```

Comment:
Nit: a little simpler?

```suggestion
  const dismissedKey = isAuthorized && userId ? getDismissedKey(userId) : GLOBAL_DISMISSED_KEY;

  useEffect(() => {
    const dismissed = localStorage.getItem(dismissedKey);
    setIsRebrandBannerActive(!dismissed);
  }, [dismissedKey]);
```

Reply:
Ooh, I like this. Updating it 🤓